geniusyield-market-maker/src/GeniusYield/MarketMaker/MakerBot.hs

```diff
@@ -115,19 +130,129 @@ buildAndSubmitActions user@User {uColl, uStakeCred} providers netId ua di = flip
     gyAwaitTxConfirmed providers awaitTxParams tid
     logInfo $ printf "Tx successfully seen on chain with %d confirmation(s)" numConfirms
 
+
+-----------------------------------------------------------------------
+-------------------------- STATE MACHINE ------------------------------
+
+mbStateMachine ∷ Strategy
+               → MakerBot
+               → GYNetworkId
+               → GYProviders
+               → PricesProviders
+               → DEXInfo
+               → MBFret
+               → StateT MBFret IO ()
+mbStateMachine runStrategy mb@MakerBot {mbUser, mbDelay, mbToken} netId providers pp di mbFret = do
+  let cfg          = ppaCommonCfg . pricesAggregatorPP $ pp
+      respiteDelay = ceiling $ pccRespiteDelayFactor cfg * fromIntegral mbDelay
+
+  case mbFret of
+    MBReady → do
+      (newActions, controller) ← lift $ runStrategy pp mbUser mbToken
+      case controller of
+        UACNormal     → do
+          lift $ buildAndSubmitActions mbUser providers netId newActions di
+
+          lift $ gyLogInfo providers logNS "---------- Done for the block! -----------"
+          lift $ threadDelay mbDelay
+
+        UACSpooked1   → do
+          put MBSpooked1 { mbs1Relax = 0, mbs1Worse = 0 }
+          lift $ gyLogInfo providers logNS "Closing all orders due to price mismatch among Prices Providers"
+
+          lift $ cancelAllOrders' mb netId providers di
+          lift $ threadDelay respiteDelay
+
+        UACSpooked2 e → do
+          put MBSpooked2 { mbs2Relax = 0 }
+          lift $ gyLogWarning providers logNS $ "Closing all orders due to: " ++ e
+
+          lift $ cancelAllOrders' mb netId providers di
+          lift $ threadDelay respiteDelay
+
+    MBSpooked1 {..} → do
+      if mbs1Relax > pccAfterExitRelaxAim1 cfg then do
+           lift $ gyLogInfo providers logNS $ "Resuming strategy"
+           put MBReady
+         else if mbs1Worse > pccAfterExitWorseMax1 cfg
+                 then do
+                   lift $ gyLogWarning providers logNS $ "Price mismatch among providers lasting a long time!"
+                   put MBSpooked2 { mbs2Relax = 0 }
+                 else do
+                   pe ← lift $ priceEstimate' pp mbToken
+
+                   case pe of
+                     PriceMismatch1    → do
+                       lift $ gyLogInfo providers logNS $ "Price mismatch persists"
+                       put MBSpooked1 { mbs1Relax = 0, mbs1Worse = mbs1Worse + 1 }
+
```

Comment:
`threadDelay` appears to be missing here. Likewise for `PriceMismatch2`, `PriceUnavailable`. Please also check for at other suitable places.

Reply:
Done; indeed `threadDelay` was handled correctly here.